### ISSUE TITLE
Implement PWA extension

### ISSUE THREAD

**Issue by Daltz333** [wpilibsuite/frc-docs#1685]
It would be totally awesome and cool to create an extension that turns Sphinx websites into PWA for offline use on mobile devices.

Basically, we should automatically add the necessary resources to our `manifest.json`. Then generate the relative metadata in the meta data of the various pages on the site.

**Comment by Daltz333** [comment]
Currently @ItayZiv and myself are investigating this. Lots of research necessary. We can use this issue to write down development notes.

**Comment by Daltz333** [comment]
Reference regarding caching for offline: https://www.simicart.com/blog/pwa-offline/

**Comment by ItayZiv** [comment]
The biggest problem would be the cache size, the entire website currently is a bit over 100MB.
One possible solution is to cache all the images in a smaller resolution (possibly only on mobile) but even so, that might not be enough for some devices. 
Regarding the 50MB limit on iOS: https://stackoverflow.com/questions/50093856/overcoming-the-50-mib-limitation-on-pwa-cache-for-ios

The other problem would be having the script to register the service worker on each page (might need a small sphinx extension)

**Comment by Daltz333** [comment]
Yeah. I definitely think we need a sphinx extension. Ideally this can scale past frc-docs and we can have the extension automatically generate the manifest.json and inject the service work into the pages (with config overrides)

**Comment by TheTripleV** [comment]
On size, if we switch from [source or responsive] to just webps, remove all the downloads, and minify the html (which I thought we already did), the whole package should come down to ~45mb which is just under the PWA limit of 50-52 MB, so pushing it.

To decrease service worker storage even more, all images can be stored in Indexeddb and be fetched on page load.

**Comment by Daltz333** [comment]
@TheTripleV the 50mb limit only applies on iOS < 13. After that, it's half of available storage. We are just moving forward without the concern of storage for now.
